web/web.go

```diff
@@ -0,0 +1,995 @@
+		return
+	}
+	if table == "" || pos == "" || newName == "" {
+		http.Error(w, fmt.Sprintf("Table name or position or new name is empty"), http.StatusBadRequest)
+	}
+	sp := sessionState.conv.SpSchema[table]
+	position, err := strconv.Atoi(pos)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting position to integer"), http.StatusBadRequest)
+		return
+	}
+	if position < 0 || position >= len(sp.Fks) {
+		http.Error(w, fmt.Sprintf("No foreign key found at position %d", position), http.StatusBadRequest)
+		return
+	}
+	if !isUniqueName(newName) {
+		http.Error(w, fmt.Sprintf("New name %s is used before in the schema", newName), http.StatusBadRequest)
+		return
+	}
+	sp.Fks[position].Name = newName
+	sessionState.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+
+func dropSecondaryIndex(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	pos := r.FormValue("pos")
+	if sessionState.conv == nil || sessionState.driver == "" {
+		http.Error(w, fmt.Sprintf("Schema is not converted or Driver is not configured properly. Please retry converting the database to spanner."), http.StatusNotFound)
+		return
+	}
+	if table == "" || pos == "" {
+		http.Error(w, fmt.Sprintf("Table name or position is empty"), http.StatusBadRequest)
+	}
+	sp := sessionState.conv.SpSchema[table]
+	position, err := strconv.Atoi(pos)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting position to integer"), http.StatusBadRequest)
+		return
+	}
+	if position < 0 || position >= len(sp.Indexes) {
+		http.Error(w, fmt.Sprintf("No secondary index found at position %d", position), http.StatusBadRequest)
+		return
+	}
+	sp.Indexes = removeSecondaryIndex(sp.Indexes, position)
+	sessionState.conv.SpSchema[table] = sp
+	updateSessionFile()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+
+// updateSessionFile updates the content of session file with
+// latest app.conv.
+func updateSessionFile() error {
+	filePath := sessionState.sessionFile
+	if filePath == "" {
+		return fmt.Errorf("Session file path is empty")
+	}
+	f, err := os.OpenFile(filePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return err
+	}
+	convJSON, err := json.MarshalIndent(sessionState.conv, "", " ")
+	if err != nil {
+		return err
+	}
+	err = f.Truncate(0)
+	if err != nil {
+		return err
+	}
+	if _, err := f.Write(convJSON); err != nil {
+		return err
+	}
+	return nil
+}
+
+// rollback is used to get previous state of conversion in case
+// some unexpected error occurs during update operations.
+func rollback(err error) error {
+	if sessionState.sessionFile != "" {
+		sessionState.conv = internal.MakeConv()
+		err2 := conversion.ReadSessionFile(sessionState.conv, sessionState.sessionFile)
+		if err2 != nil {
+			return fmt.Errorf("encountered error %w. rollback failed: %v", err, err2)
+		}
+	}
```

Comment:
I'm wondering if it's worth adding an else branch in the case where sessionFile is empty?

That situation would be unexpected and would prevent us from rolling back. An error like:

return fmt.Errorf("encountered error %w. rollback failed because we don't have a session file", err)

might be good defensive programming.

What do you think?

Reply:
Hi Nevin,

Since the err2 can be the due to the following
  1. Unavailability of file
  2. Error in reading the file, may be system call failure etc...
  3. Error due to Unmarshal

Request to kindly suggest if changes required

Reply:
Sorry I wasn't clear.
I was suggested that we generate an error when sessionFile is empty. The current code just falls through the if statement and returns err, which is potentially confusing since we've not done the rollback. Suggest something like the following, using go's early return convention:
```
func rollback(err error) error {
	if sessionState.sessionFile == "" {
		return fmt.Errorf("encountered error %w. rollback failed because we don't have a session file", err)
	}
	sessionState.conv = internal.MakeConv()
	err2 := conversion.ReadSessionFile(sessionState.conv, sessionState.sessionFile)
	if err2 != nil {
		return fmt.Errorf("encountered error %w. rollback failed: %v", err, err2)
	}
	return err
}
```

Reply:
done pl